interfaces/builtin/uinput.go

```diff
@@ -0,0 +1,67 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+// https://www.kernel.org/doc/html/latest/input/uinput.html. Manually connect
+// because this interface allows for arbitrary input injection.
```

Comment:
it seems if people in practice will have to change the udev rules to make them much more permissive this should be actually superprivileged ?

Reply:
@pedronis - are you asking that on an existing system where someone might have the udev rules already in place, would we want to make this superprivileged because it allows for (as designed of the uinput kernel interface) injecting arbitrary input? I suppose there is an argument for that, but the fact that it is manually connected and does precisely what it advertises it does, I'm leaning towards 'no'. This is an interface that we'll want to document in https://forum.snapcraft.io/t/process-for-aliases-auto-connections-and-tracks/455 so reviewers understand when voting and IME that should cover us. If you prefer it to be superprivileged, I'm happy to make that change.

Reply:
My worry is more snaps, that will both ask people to change permissions on disk and manually connect, without asking for auto-connection. If we make it superprivileged we would be in the loop also for those cases.

Reply:
@pedronis - ok, I'll make that change with a supporting comment.